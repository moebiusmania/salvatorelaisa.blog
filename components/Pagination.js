import Link from '@/components/Link'

export default function Pagination({ totalPages, currentPage }) {
  const prevPage = parseInt(currentPage) - 1 > 0
  const nextPage = parseInt(currentPage) + 1 <= parseInt(totalPages)

  return (
    <div className="pt-6 pb-8 space-y-2 md:space-y-5">
      <nav className="flex justify-between">
        {!prevPage && (
          <button rel="previous" className="cursor-auto disabled:opacity-50" disabled={!prevPage}>
            Precedente
          </button>
        )}
        {prevPage && (
          <Link href={currentPage - 1 === 1 ? `/post/` : `/post/page/${currentPage - 1}`}>
            <button rel="previous">Precedente</button>
          </Link>
        )}
        <span>
          {currentPage} di {totalPages}
        </span>
        {!nextPage && (
          <button rel="next" className="cursor-auto disabled:opacity-50" disabled={!nextPage}>
            Successivo
          </button>
        )}
        {nextPage && (
          <Link href={`/post/page/${currentPage + 1}`}>
            <button rel="next">Successivo</button>
          </Link>
        )}
      </nav>
    </div>
  )
}
