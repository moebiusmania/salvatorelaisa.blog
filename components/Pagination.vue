<script setup lang="ts">
export interface Props {
  totalPages: number;
  page: number;
  limit: number;
  allPosts: number;
}

const { totalPages, page, limit, allPosts } = defineProps<Props>();

const prev: number = page - 1;
const next: number = page + 1;
</script>

<template>
  <div class="pt-6 pb-8 space-y-2 md:space-y-5">
    <nav class="flex justify-between">
      <NuxtLink
        v-if="prev > 0"
        class="text-primary hover:text-primary-focus"
        :href="`/post/page/${prev}`"
      >
        <button aria-label="precedente">&larr; Precedente</button>
      </NuxtLink>
      <button
        aria-label="precedente"
        v-else
        disabled
        class="text-neutral opacity-40"
      >
        &larr; Precedente
      </button>

      <span>Pagina {{ page }} di {{ totalPages }}</span>

      <NuxtLink
        v-if="allPosts === limit"
        class="text-primary hover:text-primary-focus"
        :href="`/post/page/${next}`"
      >
        <button aria-label="successivo">Successivo &rarr;</button>
      </NuxtLink>
      <button
        aria-label="successivo"
        v-else
        disabled
        class="text-neutral opacity-40"
      >
        Successivo &rarr;
      </button>
    </nav>
  </div>
</template>
