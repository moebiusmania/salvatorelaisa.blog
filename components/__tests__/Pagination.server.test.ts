import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Pagination from "../Pagination.server.vue";

describe("Pagination", () => {
  // Helper function to create wrapper with common props
  const createWrapper = (page: number, totalPages: number) => {
    return mount(Pagination, {
      props: {
        page,
        totalPages,
        limit: 10,
        allPosts: totalPages * 10,
      },
      global: {
        stubs: {
          NuxtLink: {
            template: '<a :href="to"><slot /></a>',
            props: ["to"],
          },
        },
      },
    });
  };

  // Snapshot test
  it("renders correctly", () => {
    const wrapper = createWrapper(2, 5);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders page direction correctly", () => {
    const wrapper = createWrapper(2, 3);
    const links = wrapper.findAll("a");
    links.forEach((link) => {
      expect(link.text()).toBe(`← Precedente`);
    });
  });

  it("handles zero total pages gracefully", () => {
    const wrapper = createWrapper(1, 0);
    const links = wrapper.findAll("a");
    expect(links.length).toBe(0);
  });

  it("renders links in ascending order", () => {
    const wrapper = createWrapper(2, 5);
    const links = wrapper.findAll("a");
    const pageNumbers = links.map((link) => parseInt(link.text().trim()));

    for (let i = 1; i < pageNumbers.length; i++) {
      expect(pageNumbers[i]).toBeGreaterThan(pageNumbers[i - 1]);
    }
  });
});
