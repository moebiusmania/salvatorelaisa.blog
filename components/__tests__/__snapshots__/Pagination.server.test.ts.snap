// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > renders correctly 1`] = `
"<div data-v-b2d456d3="">
  <nav data-v-b2d456d3=""><a data-v-b2d456d3="" href="/post/page/1"><button data-v-b2d456d3="" aria-label="precedente">← Precedente</button></a><span data-v-b2d456d3="">Pagina 2 di 5</span><button data-v-b2d456d3="" aria-label="successivo" disabled=""> Successivo → </button></nav>
</div>"
`;
